### DISCUSSION TITLE
Avoid the danger of exception And .NET naming conventions

### DISCUSSION THREAD

**Discussion by theyasub** [CommunityToolkit/Maui#884] [Q&A] [answered]
For the BrushExtensions class, we need to make a small change.
https://github.com/CommunityToolkit/Maui/blob/main/src/CommunityToolkit.Maui.Core/Extensions/BrushExtensions.windows.cs

![image](https://user-images.githubusercontent.com/105649270/211328025-bcdb3a1f-320c-424c-a28c-014904cb298c.png)

These are the changes that need to be made:
1. Add a comment explaining the purpose of the ToColor method. This will help other developers understand what the method does and how to use it.
2. Consider renaming the method to something more descriptive, such as TryConvertToColor. This will make the purpose of the method clearer and will make the method name more consistent with the .NET naming conventions.
3. Add a null check for the brush parameter to prevent a NullReferenceException from being thrown if a null value is passed to the method.

@brminnick  

It may be a small change, but it makes our code easier to understand for any programmer who wants to read it. then I want you to pay attention to the .NET naming conventions. If you and the developer agree to this change, could you please create an issue for me?

**Comment by theyasub** [comment]
It is possible to write more understandable comments so that other programmers can easily read the code

**Comment by TimurSuyarov** [comment]
I'm with you

**Comment by VladislavAntonyuk** [comment, marked as answer]
1. This method does exactly what is described. Converts Brush to Color. How do you propose to extend it?
2. TryConvertToColor should have a different signature. You can compare int.Parse with int.TryParse for example.
3. This method will not throw null if null is passed as a parameter, because if statement returns false in that case.

**Reply by theyasub** [reply]
> 1. This method does exactly what is described. Converts Brush to Color. How do you propose to extend it?
> 2. TryConvertToColor should have a different signature. You can compare int.Parse with int.TryParse for example.
> 3. This method will not throw null if null is passed as a parameter, because if statement returns false in that case.

Ok wouldn't it be good to check just in case?
and Can I add a better and more complete comment

@brminnick

**Comment by CliffAgius** [comment]
Closed as answered
